dspace-api/src/test/java/org/dspace/discovery/DiscoveryIT.java

```diff
@@ -731,6 +737,68 @@ public void iteratorSearchServiceTest() throws SearchServiceException {
         }
     }
 
+    /**
+     * Test designed to check if default sort option for Discovery is working, using <code>workspace</code>
+     * DiscoveryConfiguration <br/>
+     * <b>Note</b>: this test will be skipped if <code>workspace</code> do not have a default sort option set and of
+     * metadataType <code>dc_date_accessioned</code> or <code>lastModified</code>
+     * @throws SearchServiceException
+     */
+    @Test
+    public void searchWithDefaultSortServiceTest() throws SearchServiceException {
+
+        DiscoveryConfiguration workspaceConf = SearchUtils.getDiscoveryConfiguration("workspace", null);
+        // Skip if no default sort option set for workspaceConf
+        if (workspaceConf.getSearchSortConfiguration().getDefaultSortField() == null) {
+            return;
+        }
+
+        DiscoverySortFieldConfiguration defaultSortField =
+                workspaceConf.getSearchSortConfiguration().getDefaultSortField();
+
+        // Populate the testing objects: create items in eperson's workspace and perform search in it
+        int numberItems = 10;
+        context.turnOffAuthorisationSystem();
+        EPerson submitter = EPersonBuilder.createEPerson(context).withEmail("submitter@example.org").build();
+        context.setCurrentUser(submitter);
+        Community community = CommunityBuilder.createCommunity(context).build();
+        Collection collection = CollectionBuilder.createCollection(context, community).build();
+        for (int i = 0; i < numberItems; i++) {
+            ItemBuilder.createItem(context, collection)
+                    .withTitle("item " + i)
+                    .build();
+        }
+
+        // Build query with default parameters (except for workspaceConf)
+        DiscoverQuery discoverQuery = SearchUtils.getQueryBuilder()
+                .buildQuery(context, new IndexableCollection(collection), workspaceConf,"",null,"Item",null,null,
+                        null,null);
+
+        DiscoverResult result = searchService.search(context, discoverQuery);
+
+        if (defaultSortField.getMetadataField().equals("dc_date_accessioned")) {
```

Comment:
Noted. I commented out the codes that test against `dc.date.accesioned`. It was a bit of a pain to figure it out and I wanted to keep it in case we have to check against it again.